tools/crm_resource_runtime.c

```diff
@@ -1798,9 +1798,15 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
     if (services_action_sync(op)) {
         exit_code = op->rc;
 
+        /* Lookup exit code based on rc for LSB resources */
+        if (pcmk__str_eq(rsc_class, PCMK_RESOURCE_CLASS_LSB, pcmk__str_casei) && 
+                pcmk__str_eq(rsc_action, "force-check", pcmk__str_casei)) {
+            exit_code = services_get_ocf_exitcode(action, exit_code);
+        }
+
```

Comment:
hmm ... would it make sense to move that exit-code-to-ocf-translation to the services-library.
execd has similar code - a little bit more elaborate maybe ...
maybe a bit in the flags-field of the action that triggers translation already inside services_action_sync ...

Reply:
I'll look into it.

Reply:
svc_action_t is public API, and the raw return code can be useful to callers, but I could see adding a new struct member for the OCF equivalent to svc_action_private_t, with a getter function. However we'd have to make sure that anytime we set the exit status, we also set the new member. I actually have some big changes pending in related functions (for a logging/status project) so I'd hold off on it for now. I hope to get back to those soon ...

Reply:
Adding a flag to public API wouldn't be an issue - right?
Question is if users wanting the translation might not at the same time like to check the original value as well - which would be lost with what I had in mind.
But well if we pick code just from execd or from here as well if we do it later doesn't make make much difference.

Reply:
A flag would be fine, but the services library itself might need the raw code, so having them both would be useful.

Also, while working on my project I found places where the library sets a synthetic exit status without knowing the resource class (we just rely on 0 being success and 1 error for all classes). svc_action_t can be used for direct execution of a binary, without a resource class, as well. Adding the OCF equivalent is doable, but it won't be trivial. Coincidentally my project involves functionizing setting rc/status so that will reduce the number of places that need to be touched.

Reply:
FYI, crm_resource doesn't need everything that the executor does -- crm_resource just supports OCF and LSB for this code.

I've had an idea in the back of my mind for object-oriented handling of resource standards. The library could gain a struct for resource standards (a class class?) that would include enum pcmk_ra_caps and methods for things like listing agents. Some of crm_resource's code could move to methods there, like all this --force-\* stuff.